java/src/Ice/src/main/java/com/zeroc/IceInternal/TcpConnector.java

```diff
@@ -22,7 +22,15 @@ public String toString() {
 
   @Override
   public int hashCode() {
-    return _hashCode;
+    int h = 5381;
+    h = HashUtil.hashAdd(h, _addr.getAddress().getHostAddress());
```

Comment:
You want to read this code together with equals and be confident `a.equals(b)` implies `a.hashCode() == b.hashCode()`. The code isc currently too different.

Reply:
Alright, I'll roll my `Network` cleanup into this PR as well then.
It should help improve the symmetry between `equals` and `hashCode`, although for the record, I'm fairly confident that we are upholding that contract here.